lib/report.js

```diff
@@ -92,14 +96,55 @@ class Report {
     for (const [converter, { path, functions }] of possibleCjsEsmBridges) {
       if (resultCountPerPath.get(path) <= 1) {
         converter.applyCoverage(functions)
-        map.merge(converter.toIstanbul())
+        map.merge(this._getIstanbulCoverageMap(converter, allFilesMap))
       }
     }
 
+    // if we are running with --all then we create an empty coverage
+    // record for any unloaded files
+    if (this.all) {
+      await this._createEmptyRecordsForUnloadedFiles(allFilesMap, map)
+    }
+
     this._allCoverageFiles = map
     return this._allCoverageFiles
   }
 
+  /**
+   * v8toIstanbul will return full paths for js files, but in cases where a sourcemap is involved (ts etc)
+   * it will return a relative path. Normally this is fine, but when using the `--all` option we load files
+   * in advance and index them by a path. Here we need to decide in advance if we'll handle full or relative
+   * urls. This function gets Istanbul CoverageMapData and makes sure all paths are relative when --all is
+   * supplied.
+   * @param {V8ToIstanbul} converter coverts v8 coverage to Istanbul's format
+   * @param {Map<string,boolean>} allFilesMap a map of files for the project that allows us to track
+   * if a file has coverage
+   * @returns {CoverageMapData}
+   * @private
+   */
+  _getIstanbulCoverageMap (converter, allFilesMap) {
```

Comment:
I wish we didn't need to do this, could we avoid this if `v8toistanbul` turned turned all paths into absolute paths.

Reply:
I think so, yes. I can make that change and test it. I'll make the other more fundamental changes around an empty v8 coverage record vs an empty istanbul coverage record first and then circle.